### ISSUE TITLE
cross entropy applied on the loss function

### ISSUE THREAD

**Issue by kaichienlogic** [kaichienlogic/logistic-regression-supported-by-the-loss-function-calculated-by-cross-entropy#1]
clc;
clear all
x=-1000:1:1000;
y=1./(1+exp(-x));

figure(1)
plot(x,y)
grid on

X=[-1000, -950, -900, 900, 950, 1000];

w=[0.001,0.002,0.005,0.0001,0.001,0.001];
tor=0.0000001;
b=1;

r=[];
Y_p=[];
Y=[];
for j=1:length(X)
    r(j)=w(j)*X(j)+b;
    Y_p(j)=1/(1+exp(-r(j)));
    Y(j)=1/(1+exp(-X(j)));
end

L=0;
count=0;
while count < length(X)
    count=count+1;
    L=L-(Y(count)*log(Y_p(count))+(1-Y(count))*log(1-Y_p(count)));
end

count=0;
while abs(L) > 0.001
    count=count+1;
    w=w-tor*((Y_p-Y))*X'/length(X);
    b=b-tor*((Y_p-Y))*ones(length(X),1)/length(X);
    
    r=[];
    Y_p=[];
    for j=1:length(X)
        r(j)=w(j)*X(j)+b;
        Y_p(j)=1/(1+exp(-r(j)));
    end
    L=0;
    c=0;
    while c < length(X)
        c=c+1;
        L=L-(Y(c)*log(Y_p(c))+(1-Y(c))*log(1-Y_p(c)));
    end
end

figure(2)
plot(X,Y_p,'bo')
hold on
plot(X,Y,'ro')
grid on
